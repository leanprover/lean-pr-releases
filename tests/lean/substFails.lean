def foo := 3
def bar := 4

def ex1  (heq : foo = bar) (P : Nat → Prop) (h : P foo)         := heq ▸ h
#check ex1
def ex2  (heq : foo = bar) (P : Nat → Prop) (h : P foo) : P 4   := heq ▸ h -- error
def ex3  (heq : foo = bar) (P : Nat → Prop) (h : P foo) : P bar := heq ▸ h
def ex4  (heq : foo = bar) (P : Nat → Prop) (h : P 3)           := heq ▸ h -- error
def ex5  (heq : foo = bar) (P : Nat → Prop) (h : P 3)   : P 4   := heq ▸ h -- error
def ex6  (heq : foo = bar) (P : Nat → Prop) (h : P 3)   : P bar := heq ▸ h

def ex7  (heq : bar = foo) (P : Nat → Prop) (h : P foo)         := heq ▸ h
#check ex7
def ex8  (heq : bar = foo) (P : Nat → Prop) (h : P foo) : P 4   := heq ▸ h -- error
def ex9  (heq : bar = foo) (P : Nat → Prop) (h : P foo) : P bar := heq ▸ h
def ex10 (heq : bar = foo) (P : Nat → Prop) (h : P 3)           := heq ▸ h -- error
def ex11 (heq : bar = foo) (P : Nat → Prop) (h : P 3)   : P 4   := heq ▸ h -- error
def ex12 (heq : bar = foo) (P : Nat → Prop) (h : P 3)   : P bar := heq ▸ h
