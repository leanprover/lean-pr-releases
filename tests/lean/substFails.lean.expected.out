ex1 (heq : foo = bar) (P : Nat → Prop) (h : P foo) : P bar
substFails.lean:6:67-6:74: error: invalid `▸` notation, expected result type of cast is 
  P 4
however, the equality 
  heq
of type 
  foo = bar
does not contain the expected result type on either the left or the right hand side
substFails.lean:8:67-8:74: error: invalid `▸` notation, the equality
  heq
has type 
  foo = bar
but neither side of the equality is mentioned in the type
  P 3
substFails.lean:9:67-9:74: error: invalid `▸` notation, expected result type of cast is 
  P 4
however, the equality 
  heq
of type 
  foo = bar
does not contain the expected result type on either the left or the right hand side
ex7 (heq : bar = foo) (P : Nat → Prop) (h : P foo) : P bar
substFails.lean:14:67-14:74: error: invalid `▸` notation, expected result type of cast is 
  P 4
however, the equality 
  heq
of type 
  bar = foo
does not contain the expected result type on either the left or the right hand side
substFails.lean:16:67-16:74: error: invalid `▸` notation, the equality
  heq
has type 
  bar = foo
but neither side of the equality is mentioned in the type
  P 3
substFails.lean:17:67-17:74: error: invalid `▸` notation, expected result type of cast is 
  P 4
however, the equality 
  heq
of type 
  bar = foo
does not contain the expected result type on either the left or the right hand side
